install-deps.sh

```diff
@@ -272,7 +272,6 @@ else
         echo "Using apt-get to install dependencies"
         $SUDO apt-get install -y devscripts equivs
         $SUDO apt-get install -y dpkg-dev
```

Comment:
see line 253 - do we want to make it optional in freebsd as well?

Reply:
i think it's up to @wjwithagen

Reply:
Yes, please.....
I'm very much in favour that all nuts and bolts glued on, are conditionalised.
Especially when they are not essential and required for the functioning of the main Ceph components.
And what I like even beter is that with this PR is that in absence of `rabbitmq` this code is now just skipped, but RGW is still build.
Also I have no way of verifying that this will work on FreeBSD, since I do not have the infrastructure that goes with it.

So in short I would not put it in the `pkg install` list for FreeBSD.

Reply:
added a commit to disable amqp endpoint for FreeBSD.

Reply:
why not make it conditional?
(BTW building was verified on freebsd12)

Reply:
@yuvalif what do you mean by conditional?

Reply:
@tchaikov i mean, based on the build flag, similar to how we do that for other OSs

Reply:
@yuvalif the cmake script handles various `WITH_*` already. on rpm based distro we have with and without conditions. on debian derivatives, i don't think we have any "build flag" aside from cmake options. probably you could help send a follow-up PR to address this? and i am happy to drop the commit disabling this feature if it makes more sense to you and @wjwithagen ?